src/app/issuer/components/badgeclass-generate-qr/badgeclass-generate-qr.component.ts

```diff
@@ -141,7 +141,7 @@ export class BadgeClassGenerateQrComponent extends BaseAuthenticatedRoutableComp
 				//@ts-ignore
 				(qrCode.valid_from && !isNaN(new Date(qrCode.valid_from)))
 			) {
-				if (new Date(this.valid_from) < new Date() && new Date(this.expires_at) >= new Date()) {
+				if (new Date(this.valid_from) < new Date() && new Date(this.expires_at).toDateString() >= new Date().toDateString()) {
```

Comment:
@jona159 I just came across this due to curiosity. This looks deeply unsettling to me. ;-) Are you comparing string lengths as a replacement for date comparison?

Reply:
Not the length of the string, but still... maybe not the best solution. Doing real date comparison now

Reply:
@jona159 Then something like byte lenght? `>=` does not do any semantics with strings. Example:

```
const event1 = new Date(1993, 6, 28, 14, 39, 7);
const event2 = new Date(1993, 7, 1, 13, 40, 8);

console.log(event1.toString());
console.log(event2.toString());

console.log(event1.toDateString());
console.log(event2.toDateString());

console.log(event1.toDateString() >= event2.toDateString())
// evaluates to "true"
```

Reply:
It was doing ```"lexicographical comparison"```, so **comparing the characters in a string sequentially from left to right**. Which means it would evaluate ``` 12/25/2023' > '3/15/2024'``` as true, because 12 > 3